massa-network/src/peer_info_database.rs

```diff
@@ -494,50 +460,35 @@ impl PeerInfoDatabase {
         if !ip.is_global() {
             return Err(NetworkError::InvalidIpError(*ip));
         }
-        let (available_bootstrap_conns, available_nonbootstrap_conns) =
-            self.get_available_out_connection_attempts();
+        let mut update_happened = false;
+        let peer_type = if let Some(p) = self.get_peer_type(ip) {
+            p
+        } else {
+            let mut p = PeerInfo::new(*ip, false);
+            p.active_out_connection_attempts += 1;
+            self.peers.insert(*ip, p);
+            update_happened = true;
+            self.get_peer_type(ip).unwrap()
+        };
+
+        self.update_global_active_out_connection_attempt_count(
```

Comment:
here is a logical problem: if there are no more out connection slots, this call will bail, but the peer could already have been created + inserted into self.peers (see "else" branch just above), which will leave the state inconsistent

Reply:
How much is it critical, if the local counter of the peer is not updated, and we just keep the ip in the db ?

Reply:
the counters were inconsistent. Now you added a rollback mechanism, but I'm not a big fan of this.

Would be nicer to always first test if we can insert/update a peer BEFORE actually doing it. Also elsewhere in this PR